gpage.h

```diff
@@ -222,16 +228,17 @@ namespace gcpp {
 	//
 	inline
 	bool gpage::contains(gsl::not_null<const byte*> p) const noexcept {
-		return &storage[0] <= p && p < &storage[total_size - 1];
+		auto cmp = std::less<>{};
+		return !cmp(p, &storage[0]) && cmp(p, &storage[total_size - 1]);
```

Comment:
In passing, this block of code looks like it would be clearer (and possibly safer) if you used a `span` over `storage`. That way you could encapsulate+enforce the `total_size-1` assumption (if it is still desired).